mirrord/cli/src/wizard.rs

```diff
@@ -0,0 +1,671 @@
+            ))
+            .layer(SetResponseHeaderLayer::overriding(
+                header::EXPIRES,
+                header::HeaderValue::from_static("0"),
+            ))
+            .service(ServeFile::new(temp_dir_path.join("index.html")));
+
+        ServiceBuilder::new()
+            .layer(SetResponseHeaderLayer::if_not_present(
+                header::CACHE_CONTROL,
+                header::HeaderValue::from_static("public, max-age=1209600"),
+            ))
+            .service(
+                ServeDir::new(temp_dir_path)
+                    .fallback(index_service)
+                    .append_index_html_on_directories(false),
+            )
+    };
+
+    // prepare temp dir as SPA with endpoints behind `/api` path
+    let app = Router::new()
+        .layer(TraceLayer::new_for_http())
+        .fallback_service(serve_client)
+        .route(
+            "/api/v1/is-returning",
+            get(|| async { is_returning_string }),
+        )
+        .route("/api/v1/cluster-details", get(cluster_details))
+        .route("/api/v1/namespace/{namespace}/targets", get(list_targets))
+        .with_state(Arc::new(Mutex::new((user_data, client))));
+    let addr = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), 3000);
+    let listener = tokio::net::TcpListener::bind(addr).await?;
+    tracing::debug!("listening on {}", listener.local_addr()?);
+    progress.success(None);
+
+    parent_progress
+        .subtask("<|:) Greetings, traveler!")
+        .success(None);
+
+    // open browser window
+    let url = "http://localhost:3000/";
+    match opener::open(url) {
+        Ok(()) => {}
+        Err(error) => {
+            tracing::trace!(?error, "failed to open browser");
+            parent_progress
+                .subtask("Couldn't open the browser automatically")
+                .failure(None);
+        }
+    }
+    parent_progress
+        .subtask(format!("The wizard is available at: {url}").as_str())
+        .success(None);
+
+    parent_progress.success(None);
+    axum::serve(listener, app.layer(TraceLayer::new_for_http()))
+        .await
+        .expect("Infallible operation should never fail");
+
+    Ok(())
+}
+
+/// The response returned from the `/api/v1/cluster-details` endpoint with namespaces and
+/// target_types that mirrord can target in the user's cluster.
+#[derive(Debug, Serialize)]
+struct ClusterDetails {
+    namespaces: Vec<String>,
+    target_types: Vec<TargetType>,
+}
+
+/// Represents the relevant information for each available target returned by the
+/// `/api/v1/namespace/{namespace}/targets` endpoint. `detected_ports` are used in port
+/// configuration in the wizard's Network tab to offer ports that the user is most likely to be
+/// interested in, according to the ports exposed by that target's Pod or Pods
+#[derive(Debug, Serialize)]
+struct TargetInfo {
+    target_path: String,
+    target_namespace: String,
+    detected_ports: Vec<u16>,
```

Comment:
```suggestion
/// - Shown in the wizard's `Network` tab
///
/// Ports exposed by the target that we have auto-detected.
///
/// These are the ports the user is most likely to be interested in.
detected_ports: Vec<u16>,
```

Something like this, and then remove this part of the docs from the struct itself, wdyt?